packages/storefront-events-sdk/src/Base.ts

```diff
@@ -33,6 +33,12 @@ export abstract class Base {
 
     // Push event to ACDL
     protected pushEvent(event: EventName, context: CustomContext = {}): void {
+        // if param context.customContext was set as the default by publishManager, it will be removed
+        // so that a separately set custom context can be applied
+        // otherwise param context.customContext will override previously set custom context
+        if (context && context.customContext === undefined) {
```

Comment:
i'm pretty sure this is always going to be false. even if context was passed in as null or undefined or any falsey, it's being defaulted to an empty object which is not going to evaluate to to since an empty object is not undefined.

more importantly, if this line ever was ever true, it would actually delete the whole event and push nothing to the acdl even for non-custom event

Reply:
@grahamcrackers not sure that I understand you correctly. Publish manager adds this context to all events, see https://github.com/adobe/commerce-events/blob/main/packages/storefront-events-sdk/src/PublishManager.ts#L61
in most cases it will have value `{ customContext: context }`, where **context** is undefiled.
so, we check that customContext is exactly undefined it means that customer set nothing in the push event method (e.g. `pageView()`). Customer could set empty object in the push method (e.g. `pageView({})` ) to override previously set custom context (which was set to ACDL)

Reply:
more with examples
```
const mse = window.magentoStorefrontEvents;

mse.context.setCustom({
    commerce: {
        saveForLaters: {
            value: 1,
        },
    },
});
mse.publish.pageView();
```
without that change, previously set custom context will be overwritten with empty value in [event info](https://github.com/adobe/commerce-events/pull/160/files#diff-ad52ce934b813892d2be72ce61bcf5016b7d7d5b17f19293190d14644d9268beR43-R49), and [handler will have customContext as undefined](https://github.com/adobe/commerce-events/blob/main/packages/storefront-events-collector/src/handlers/page/viewAEP.ts#L10),

Reply:
actually... yep, you are right. sorry. I was thinking `context.customContext` was passed in as the context, and not that the `customContext` was the child on the `context`...

These types are a bit confusing.